VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitVideoFragment.java

```diff
@@ -272,33 +272,125 @@ private void startOnlineDownload(DownloadEntry videoData, ProgressWheel progress
     }
 
 
+    public synchronized void playVideoModel(final DownloadEntry video) {
+        try {
+            if (playerFragment.isPlaying()) {
+                if (video.getVideoId().equals(playerFragment.getPlayingVideo().getVideoId())) {
+                    logger.debug("this video is already being played, skipping play event");
+                    return;
+                }
+            }
+        } catch(Exception ex) {}
+
+
+        try{
+
+            // reload this model
+            new Storage(getActivity()).reloadDownloadEntry(video);
+
+            logger.debug("Resumed= " + playerFragment.isResumed());
+            if ( !playerFragment.isResumed()) {
+                // playback can work only if fragment is resume
+                if (playPending != null) {
+                    playHandler.removeCallbacks(playPending);
+                }
+                playPending = new Runnable() {
+                    public void run() {
+                        playVideoModel(video);
+                    }
+                };
+                playHandler.postDelayed(playPending, 200);
+                return;
+            } else {
+                if (playPending != null) {
+                    playHandler.removeCallbacks(playPending);
+                }
+            }
+
+
+
+            TranscriptModel transcript = null;
+            if(unit!=null && unit.getVideoResponseModel() != null) {
+                if (unit.getVideoResponseModel().getSummary() != null) {
```

Comment:
Should this just be another ``&&`` clause instead of being a separate nested ``if``?